package/MDAnalysis/auxiliary/EDR.py

```diff
@@ -0,0 +1,358 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+"""
+EDR auxiliary reader --- :mod:`MDAnalysis.auxiliary.EDR`
+========================================================
+.. versionadded:: 2.3.0
+
+EDR_ files are binary files following the XDR_ protocol. They are written by
+GROMACS during simulations and contain the time-series energy data of the
+system.
+
+pyedr_ is a Python package that reads EDR binary files and returns them
+human-readable form as a dictionary of NumPy arrays. It is used by the EDR
+auxiliary reader to parse EDR files. As such, a dictionary with string keys and
+numpy array values is loaded into the :class:`EDRReader`. It is basically a
+Python-based version of the C++ code in GROMACS_.
+
+The EDR auxiliary reader takes the output from pyedr and loads the energy data
+as auxiliary data into :class:`MDAnalysis.core.universe.Universe`. Standalone
+usage is also possible, where the energy terms are extracted without
+associating them with the trajectory, for example, to allow easy plotting of
+the energy terms.
+
+For adding EDR data, 2-3 arguments need to be provided to
+:func:`MDAnalysis.coordinates.base.add_auxiliary`: `auxname`, `auxdata`, and
+(optionally) `auxterm`. `auxterm`s, if provided need to exactly match the names
+of EDR terms to be added. When an auxterm is provided, `auxname` can be freely
+chosen. If an `auxterm` is not provided, `auxname` must instead exactly match
```

Comment:
This is a little bit confusing, why not make `auxname` optional and make `auxterm` the thing that should always match the EDR terms?

Reply:
I agree that this isn't ideal, I'm not very happy with this solution... The problem here is that add_auxiliary() is also used to add data from XVGReader instances, which do not have the equivalent of `auxterm`, and where `auxname` is freely assignable. I did it this way for now so using XVGReader works the same as before. But yeah, it is a confusing mess as it is now, so it needs to be changed one way or another. One solution would be reader-specific `add_auxiliary` methods, but I think that would also complicate usage of the system and make it less agnostic to file formats. 
I believe some compromise needs to be made here - We could make both `auxname` and `auxterm` initially "optional", for example, and then when the method has determined what kind of AuxReader it is dealing with, raise exceptions if for example an `auxterm` isn't provided or does not match any terms in the file? But I guess then users would need to call it as `add_auxiliary(auxterm=something)` explicitly.